lib/workers/qemu.js

```diff
@@ -57,9 +50,22 @@ module.exports = class QEMUWorker {
       '-smp', '4'
     ])
 
-    childKillEventHandler(this.child, 'exit')
-    childKillEventHandler(this.child, 'uncaughtException')
-    childKillEventHandler(this.child, 'unhandledRejection')
+    process.once('exit', (code) => {
+      this.child.kill()
+      process.exitCode = code || 0
+    })
+
+    process.once('unhandledRejection', (err) => {
```

Comment:
Why are we handling this globally? If there was an unhandled rejection, then that is probably pointing out a legit place where a promise can be thrown that we're not paying attention to. Can we deal with it at the source?

Reply:
During our setup we rejected promises mean that we have to stop execution, we can catch them, log and then exit manually, but we always handled rejections at a global level (it was ava that was doing it, but nevertheless). 
Might be worth changing the pattern

Reply:
What you give me a more concrete example of a promise that is rightfully rejected? ie a line number?

Reply:
If a device does not come online, this will reject:  https://github.com/resin-io/resinos-tests/blob/master/lib/main.js#L109.
If the right login is not provided this, this will reject: https://github.com/resin-io/resinos-tests/blob/master/lib/main.js#L65

Reply:
@telphan OK, so what we need to do is add a `.catch()` block here: https://github.com/resin-io/resinos-tests/blob/master/lib/main.js#L220 with the corresponding logic, and then the global handler should become unnecessary.